actions/platform/tam/tam_request.py

```diff
@@ -0,0 +1,206 @@
+        domain: DomainDict,
+    ) -> Dict[Text, Any]:
+        return {_TAM_CONFIRMATION: slot_value}
+
+    async def run(
+        self, dispatcher: CollectingDispatcher, tracker: Tracker, domain: Dict
+    ) -> List[Dict[Text, Any]]:
+        events = await super().run(dispatcher, tracker, domain)
+        requested_slot = tracker.get_slot("requested_slot")
+
+        if requested_slot == _TAM_DURATION:
+            dispatcher.utter_message(response="utter_access_request_tam_roles_note")
+
+        return events
+
+
+class ExecuteTAMRequest(Action):
+    def name(self) -> Text:
+        return "execute_form_access_request_tam"
+
+    async def run(
+        self, dispatcher: CollectingDispatcher, tracker: Tracker, domain: Dict
+    ) -> List[Dict[Text, Any]]:
+        account_id = tracker.get_slot(_TAM_ACCOUNT_ID)
+        org_id = tracker.get_slot(_TAM_ORG_ID)
+        duration = tracker.get_slot(_TAM_DURATION)
+        confirmation = tracker.get_slot(_TAM_CONFIRMATION)
+
+        if not confirmation:
+            dispatcher.utter_message(response="utter_access_request_tam_stopped")
+            return []
+
+        start_date, end_date = get_start_end_date_from_duration(duration)
+
+        response, content = await get_roles_for_tam(tracker)
+        if not response.ok or "data" not in content:
+            dispatcher.utter_message(response="utter_access_request_tam_error")
+            return []
+
+        roles = format_roles_list_for_tam(content["data"])
+
+        body = format_access_request_tam(
+            account_id, org_id, start_date, end_date, roles
+        )
+
+        response = await send_rbac_tam_request(tracker, body)
+        if not response.ok:
+            dispatcher.utter_message(response="utter_access_request_tam_error")
+            return []
+
+        dispatcher.utter_message(response="utter_access_request_tam_success")
+
+        return []
+
+
+async def get_roles_for_tam(tracker):
+    return await send_console_request(
+        "rbac",
+        "/api/rbac/v1/roles/?system=true&limit=9999&order_by=display_name&add_fields=groups_in_count",
+        tracker,
+        method="get",
+        fetch_content=True,
+    )
+
+
+def format_roles_list_for_tam(data: List[Dict[str, Any]]):
+    names = []
+    for role in data:
+        names.append(role.get("name"))
+    return names
+
+
+def format_access_request_tam(
+    account_id: str, org_id: str, start_date: str, end_date: str, roles: List[str]
+):
+    return {
+        "target_account": account_id,
+        "target_org": org_id,
+        "start_date": start_date,
+        "end_date": end_date,
+        "roles": roles,
+    }
+
+
+async def send_rbac_tam_request(tracker: Tracker, body: Dict[str, Any]):
+    if app.is_running_locally:
+        print("called send_rbac_tam_request in local envionment with body: ", body)
+
+        from unittest.mock import Mock
+
+        mock_response = Mock()
+        mock_response.ok = True
+
+        return mock_response
```

Comment:
This is interesting!

Reply:
Google helped! I was trying to copy the access request flow but also check the status :)